torch/csrc/autograd/variable.h

```diff
@@ -31,8 +31,12 @@ struct Variable : public at::Tensor {
 
   // Implicitly casts a Tensor to a Variable. This should only be called on
   // Tensors which you know are actually Variables.
-  /*implicit*/ Variable(Tensor const & rhs) : Tensor(rhs) {}
-  /*implicit*/ Variable(Tensor && rhs) noexcept : Tensor(std::move(rhs)) {}
+  /*implicit*/ Variable(Tensor const & rhs) : Tensor(rhs) {
+    AT_ASSERT(rhs.isVariable(), "not variable");
+  }
+  /*implicit*/ Variable(Tensor && rhs) noexcept : Tensor(std::move(rhs)) {
+    AT_ASSERT(rhs.isVariable(), "not variable");
```

Comment:
I think these should be `isVariable()` (i.e. `self->isVariable()`). I think `rhs` will be undefined after the move construction.

Might as well make the above constructor do the same thing too.